extensions/amp-sidebar/0.1/amp-sidebar.js

```diff
@@ -218,6 +217,55 @@ export class AmpSidebar extends AMP.BaseElement {
     );
   }
 
+  /**
+   * Gets a close button, provided by the page author, if one exists.
+   * @return {?Element} The close button.
+   */
+  getExistingCloseButton_() {
+    const candidates = this.element.querySelectorAll('[on]');
+
+    for (let i = 0; i < candidates.length; i++) {
+      const candidate = candidates[i];
+      const hasAction = this.action_.hasResolvableActionForTarget(
+        candidate,
+        'tap',
+        this.element,
+        candidate.parentNode
+      );
+
+      if (hasAction) {
+        return candidate;
+      }
+    }
+
+    return null;
+  }
+
+  /**
+   * Creates an "invisible" close button for screen readers to close the
+   * sidebar.
+   * @return {!Element}
+   */
+  createScreenReaderCloseButton() {
+    // Replacement label for invisible close button set value in amp sidebar
+    const ariaLabel =
+      this.element.getAttribute('data-close-button-aria-label') ||
+      'Close the sidebar';
+
+    // Invisible close button at the end of sidebar for screen-readers.
+    const screenReaderCloseButton = this.document_.createElement('button');
+
+    screenReaderCloseButton.textContent = ariaLabel;
+    screenReaderCloseButton.classList.add('i-amphtml-screen-reader');
+    // This is for screen-readers only, should not get a tab stop.
+    screenReaderCloseButton.tabIndex = -1;
```

Comment:
I'm not sure I understand this line. How can a screen reader user select the button and click it if has `tabIndex="-1"`?